File: backend-v2/src/main/java/vn/io/vutiendat3601/shop/v2/product/ProductDto.java
package vn.io.vutiendat3601.shop.v2.product;

import java.math.BigDecimal;
import java.util.List;

public record ProductDto(
    String productNo,
    String sku,
    String name,
    String description,
    BigDecimal unitPrice,
    BigDecimal unitListedPrice,
    String thumbnail,
    Long buyedCount,
    List<String> tags,
    Long likedCount,
    Boolean isActive,
    Integer unitsInStock,
    String categoryCode) {}

File: frontend-v2/src/app/domain/product/product-dto.ts
export class ProductDto {
  constructor(
    public productNo: string,
    public sku: string,
    public slug: string,
    public name: string,
    public description: string,
    public unitPrice: number = 0,
    public unitListedPrice: number = 0,
    public thumbnail: string | null,
    public buyedCount: number = 0,
    public tags: string[],
    public likedCount: number = 0,
    public unitsInStock: number,
    public categoryCode: string,
    public isActive: boolean = true
  ) {}
}
